pyuvdata/uvbeam/uvbeam.pyx

```diff
@@ -53,18 +53,30 @@ cpdef numpy.ndarray[dtype=numpy.npy_bool] find_healpix_indices(
       for itheta in range(n_theta):
           if found_pixel:
               break
+          #  positive and negative latitudes are actually identical
+          #  since the origin of the coordinate system is a Pole
           theta_g = theta_grid[itheta]
+
+          if theta_g < 0:
+            theta_g = fabs(theta_g)
+
           dtheta = fabs(theta_h - theta_g)
-          if dtheta > PI:
+          while dtheta > PI:
               dtheta -= 2 * PI
```

Comment:
I think this should maybe be `dtheta -= PI` because theta only runs 0 -> pi.

Reply:
@plaplant Matt wanted your opinion on this.

Reply:
Yes, I was just chatting with Matt about this. I think we need to change the logic slightly, because while it's true that positive and negative values of theta map to the same latitudes, they do not map to the same values of phi. In particular, a negative value (-theta, phi) maps to (+theta, phi + pi). See [the wiki page on spherical coordinates](https://en.wikipedia.org/wiki/Spherical_coordinate_system#Unique_coordinates).

I think the approach here should be: (1) normalize the value of dtheta to map to 0 <= dtheta <= pi, adding phase shifts to phi as necessary, then (2) normalize the value of dphi to map to 0 <= dphi <= 2pi. That should make it so our distance tests are working correctly for arbitrary input. I can try to implement this (though probably won't happen today).